edx_toggles/toggles.py

```diff
@@ -0,0 +1,60 @@
+from abc import ABC
+
+
+from django.conf import settings
+
+
+class BaseToggle(ABC):
+    """
+    This abstract base class exposes the basic API required by toggle classes.
+    """
+
+    def __init__(self, name, default=False):
+        self.name = name
+        self.default = default
+
+    def is_enabled(self):
+        raise NotImplementedError
+
+
+class ClassInstancesMixin:
+    """
+    This mixin makes it possible to track the toggle instances. To do so, call the `.create()`
+    method::
+
+        class MyToggle(ClassInstancesMixin):
+            ...
+
+        my_toggle = MyToggle.create(..., module_name=__name__)
+    """
+
+    # Note that we will always on this dict by sorted keys, so it would make sense to use a sortedcontainer here, but
+    # it would introduce an additional dependency in this package
+    _class_instances = dict()
+
+    @classmethod
+    def create(cls, *args, module_name="", **kwargs):
```

Comment:
I now understand that this is a factory method.  But, I still don't see the value it adds, and would prefer to make things work like WaffleFlag.

You wrote:
>  ```
>  MY_TOGGLE = SettingToggle.add("my_toggle", default=False, module_name=__name__)
> ```
> And then, we don't need to pass the module_name when checking whether a feature toggle is enabled:

But, WaffleFlag doesn't need to pass `module_name` when checking if it is enabled.

Here is an example declaration and usage from edx-platform.  You can find them here: https://github.com/edx/edx-platform/search?q=HIDE_ACCESS_DENIALS_FLAG

**Example Declaration:**
```
HIDE_ACCESS_DENIALS_FLAG = WaffleFlag(
    waffle_namespace=COURSE_BLOCKS_API_NAMESPACE,
    flag_name=u'hide_access_denials',
    module_name=__name__,
)
```

**Example Usage:**
```
if HIDE_ACCESS_DENIALS_FLAG.is_enabled():
    ...
```

Reply:
1. I’d refactor that code to provide a constant for the instance.
2. The toggle state endpoint is also available on Open edX instances, and the module name on its own is included in the results For each toggle. Additionally, edX will use this to derive code_owner, but that is not its only purpose.
I still prefer it on the constructor.

Reply:
:ok_hand: I'll get rid of the `create` method. As a consequence I'll also change the `_class_instances` attribute in a `WeakSet`, similar to the waffle_utils classes.